modules/statistics/php/NDB_Form_statistics.class.inc

```diff
@@ -545,19 +545,56 @@ class NDB_Form_Statistics extends NDB_Form
         } else {
             $Param_Project = '';
         }
-        $subprojects = Utility::getSubprojectsForProject($currentProject);
+        $subprojects[null] = 'All Valid Sub-Projects';
+
+        foreach (Utility::getSubprojectsForProject($currentProject)
+            as $key=>$value) {
+            $subprojects[$key] = $value;
+        }
 
-        $this->tpl_data['Subprojects'] = $subprojects;
         foreach ($subprojects as $key=>$val) {
             $subprojList .= $key.",";
         }
-        $subprojList = substr($subprojList, 0, -1);
-        if (!empty($subprojList)) {
-            $suproject_query ="AND s.SubprojectID IN ($subprojList)";
+
+        $this->tpl_data['Subprojects'] = $subprojects;
+        $currentSubproject = null;
+        $subprojList       = substr($subprojList, 1, -1);
+        if ($_REQUEST['BehaviouralSubproject'] ?? ''
+            && in_array(
+                $_REQUEST['BehaviouralSubproject'],
+                array_keys($subprojects),
+                true
+            )
+        ) {
+            $BS = 'BehaviouralSubproject';
+            $currentSubproject = htmlspecialchars($_REQUEST[$BS]);
+            $this->tpl_data['CurrentSubproject']
+                = array(
+                   'ID'   => $currentSubproject,
+                   'Name' => $subprojects[$currentSubproject],
+                  );
+            $subproject_query     = "AND (s.SubprojectID IS NULL";
+            $subproject_query    .= " OR s.SubprojectID=:spid)";
+            $this->params['spid'] =$currentSubproject;
         } else {
-            $suproject_query ='';
+            $subproject_query ="";
+        }
+        if ($_REQUEST['BehaviouralSubproject'] ?? ''
+            && !in_array(
+                $_REQUEST['BehaviouralSubproject'],
+                array_keys(
+                    $subprojects,
+                    true
+                )
```

Comment:
I think I'm missing something.. isn't this just an if/else with the inverse logic of the if/else directly above it? Why not just add it to the existing code blocks?